apps/resources/views.py

```diff
@@ -172,3 +175,34 @@ def get_context_data(self, *args, **kwargs):
         )
         context['featured_resources'] = featured_resources
         return context
+
+
+@require_POST
+@staff_member_required
+@permission_required('resources.change_resource')
+def admin_categorise_resources_view(request):
+    """
+    Admin Action for assigning categories to resources.
+    """
+    resources = Resource.objects.filter(id__in=request.POST.get('resource_ids', '').split(','))
+    categories = ResourceCategory.objects.filter(id__in=request.POST.getlist('category_ids', []))
+    remove_categories = request.POST.get('remove') == '1'
+
+    # Ensure permissions - this shouldn't be needed unless someone is being malicious:
+    if not request.user.is_superuser and request.user.approved_organisations.exists():
+        resources = resources.filter(organisation__in=request.user.approved_organisations.all()
+                                     ).distinct()
+
+    if categories:
+        for resource in resources:
```

Comment:
think this nested loop is all good as long as we're not expecting huge selections

Reply:
Is there a django query to do mass manytomany connection magic?  It would be nice.
I tested it with all the resources in the system now (500+) and it works OK speedwise...  it shouldn't be a common operation anyway.

Reply:
Yeah, it's better to do a bulk change, really.